### PR TITLE
Add dropout for tensorflow frontend

### LINKED ISSUE #4475: dropout

### PR DESCRIPTION
**Pull request by Change-m**
Close #4475

### CODE REVIEW

**Comment by VedPatwardhan** [comment]
I'll close this PR as discussed. Thanks 🙂

File: ivy/functional/frontends/tensorflow/nn.py [hunk 1]
```diff
@@ -123,3 +123,7 @@ def batch_normalization(x, mean, variance, offset, scale, variance_epsilon, name
     return x * ivy.astype(inv, x.dtype, copy=False) + ivy.astype(
         offset - mean * inv if offset is not None else -mean * inv, x.dtype
     )
+
+
+def dropout(x, rate, noise_shape=None, seed=None, name=None):
+    return ivy.dropout(x, rate, noise_shape, seed)
```
Comment (VedPatwardhan) [inline ivy/functional/frontends/tensorflow/nn.py:129]: The parameters of the `ivy.dropout` function don't match with the actual parameters accepted by the function, please check the `dropout` function inside `functional/ivy/layers.py` and pass the parameters accordingly. Firstly, the parameter names of the `ivy.dropout` function may not always match with the parameters of the frontend function, but in such cases you can just refer the docs of the tensorflow dropout function and the docstring of the `ivy.dropout` function and understand which parameters have the same meaning. Finally, when you pass the parameters, there are some that are positional-only (the ones before `/`) and some that are keyword-only (the ones after `*`) so please pass them accordingly.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @Change-m, just requested some changes, feel free to re-request my review once you're done with them. Thanks 🙂

**Comment by Change-m** [comment]
Hi @VedPatwardhan, I do the suggested changes, Please review.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @Change-m, there are a few syntax errors in your PR, for e.g. numpy is being used but isn't imported, etc. Also, your PR is 532 commits behind master as the helpers have changed a lot, please pull the latest code on master, run the test for your function and resolve all syntax errors which occur as a result. 
Feel free to re-request my review once you're done with the same, also please use the right pane to request a review so that I get notified immediately about it. 
Let me know if you have any questions. Thanks 🙂

**Review by VedPatwardhan — APPROVED** [review]
lgtm!

**Pull request merged**